packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -221,6 +221,28 @@ type ManyToManyContext = {
   directive: CodeGenDirective;
 };
 
+// Configuration used for processing directives
+export type CodeGenDirectiveProcessConfig = {
+  // This flag is going to be used for using custom primary key feature
+  isCustomPKEnabled: boolean,
```

Comment:
does this mean that there is a `@primaryKey` present in the schema or just that if it's present, we will codegen differently?

Reply:
This is determined by the feature flag value rather than the presence of the directive. If the value is true and the schema is a CPK (the @primaryKey is attached to non-id field/id PK with SK defined), it will codegen differently